grants_tagger/models/mesh_tfidf_svm.py

```diff
@@ -82,7 +89,11 @@ def fit(self, X, Y):
             X_vec = self.vectorizer.transform(X)
             self.classifier.fit(X_vec, Y[:, tag_i : tag_i + self.y_batch_size])
 
-            # TODO: Sparsify weights before saving
+            for i, est in enumerate(self.classifier.estimators_):
+                est.coef_[np.abs(est.coef_) < self.cutoff_prob] = 0
+                est.sparsify()
+                self.classifier.estimators_[i] = est
+
             with open(f"{self.model_path}/{tag_i}.pkl", "wb") as f:
                 f.write(pickle.dumps(self.classifier))
```

Comment:
Is there a reason for using `pickle.dumps()` rather than `pickle.dump`. If you use the latter, you don't need `f.write()`:

```
            with open(f"{self.model_path}/{tag_i}.pkl", "wb") as f:
                pickle.dump(self.classifier, f)
```

> pickle.dump(obj, file, protocol=None, *, fix_imports=True, buffer_callback=None)¶
> Write the pickled representation of the object obj to the open file object file. This is equivalent to Pickler(file, ?protocol).dump(obj).
>
> Arguments file, protocol, fix_imports and buffer_callback have the same meaning as in the Pickler constructor.
>
> Changed in version 3.8: The buffer_callback argument was added.

https://docs.python.org/3/library/pickle.html

Reply:
I am defaulting to this way for a while as its easier to remember for me. In other cases I was always confusing whether it was dump or dumps. Now I only use `loads` and `dumps` explicitly